nemo/collections/nlp/modules/common/transformer/transformer_generators.py

```diff
@@ -188,7 +190,7 @@ def __init__(self, embedding, decoder, log_softmax, beam_size=1, temperature=1.0
         self.beam_size = beam_size
         self.temp = temperature
 
-    @torch.no_grad()
+    # @torch.no_grad()
```

Comment:
just wondering - why these no_grad() are no longer needed?

Reply:
I was thinking that weights should be frozen anyway during eval, so that these calls would be redundant.

Reply:
PTL calls .eval() and .no_grad() for val and test: https://pytorch-lightning.readthedocs.io/en/stable/converting.html#find-the-val-loop-meat

Reply:
What about using these classes outside of PTL context?